plugins/coreboot/fu-coreboot-fmap.c

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (C) 2019 9elements Agency GmbH <patrick.rudolph@9elements.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <string.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/mman.h>
+#include <glib.h>
+#include <glib/gstdio.h>
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-device-metadata.h"
+#include "fu-device-private.h"
+#include "fu-plugin-coreboot.h"
+#include "fmap_serialized.h"
+
+/* Depends on the kernel module coreboot-fmap.ko */
+
+/*
+ * Provides the FMAP through a kernel module if available.
+ * The caller has to free the returned object.
+ */
+struct fmap *
+fu_plugin_coreboot_find_fmap (FuPlugin *plugin, GError **error)
+{
+	gchar *tmp = NULL;
+
+	if (!g_file_get_contents ("/sys/firmware/fmap", &tmp, NULL, error)) {
+		g_prefix_error (error, "Failed to read /sys/firmware/fmap: ");
+		return NULL;
+	}
+
+	// FIXME: Do sanity check
+	return (struct fmap *)tmp;
+}
+
+/*
+ * Provides the active CBFS partition through a kernel module if available.
+ * The caller has to free the returned object.
+ */
+gchar *
+fu_plugin_coreboot_find_cbfs_active_partition (FuPlugin *plugin, GError **error)
+{
+        gchar *tmp = NULL;
+
+        if (!g_file_get_contents ("/sys/firmware/cbfs_active_partition", &tmp, NULL, error)) {
+		g_prefix_error (error, "Failed to read /sys/firmware/cbfs_active_partition: ");
+                return NULL;
+	}
+
+        // FIXME: Do sanity check
+        return tmp;
+}
+
+/* FIXME: The following only works on some x86 platforms that have STRICT_DEV_MEM disabled */
```

Comment:
I would say this is incredibly important to come up with something else.  `STRICT_DEV_MEM`  has been default since 4.16.

Reply:
needs to be evaluated how to do it. the information could be read by coreboot at boot time and exposed via additional kernel modules.
The question is, do we need the partition's coreboot version at all?
We have the current version from dmi tables already.
TODO for me: Describe possible partition schemes and update methods, possible failures and recovery mechanisms.